crates/sui-framework/sources/test/random.move

```diff
@@ -0,0 +1,400 @@
+            let byte = (vector::pop_back(&mut bytes) as u64);
+            value = value + (byte << 8*i);
+            i = i + 1;
+        };
+        value
+    }
+
+    /// Use the given pseudo-random generator and a non-zero `upper_bound` to generate a
+    /// random `u64` integer in the range [0, ..., upper_bound - 1]. Note that if the upper
+    /// bound is not a power of two, the distribution will not be completely uniform.
+    public fun next_u64_in_range(random: &mut Random, upper_bound: u64): u64 {
+        assert!(upper_bound > 0, 0);
+        next_u64(random) % upper_bound
+    }
+
+    /// Use the given pseudorandom generator to generate a random `u32`.
+    public fun next_u32(random: &mut Random): u32 {
+        let bytes = next_digest(random);
+        (vector::pop_back(&mut bytes) as u32)
+            + ((vector::pop_back(&mut bytes) as u32) << 8)
+            + ((vector::pop_back(&mut bytes) as u32) << 16)
+            + ((vector::pop_back(&mut bytes) as u32) << 24)
+    }
+
+    /// Use the given pseudo-random generator and a non-zero `upper_bound` to generate a
+    /// random `u32` integer in the range [0, ..., upper_bound - 1]. Note that if the upper
+    /// bound is not a power of two, the distribution will not be completely uniform.
+    public fun next_u32_in_range(random: &mut Random, upper_bound: u32): u32 {
+        assert!(upper_bound > 0, 0);
+        next_u32(random) % upper_bound
+    }
+
+    /// Use the given pseudorandom generator to generate a random `u16`.
+    public fun next_u16(random: &mut Random): u16 {
+        let bytes = next_digest(random);
+        (vector::pop_back(&mut bytes) as u16)
+            + ((vector::pop_back(&mut bytes) as u16) << 8)
+    }
+
+    /// Use the given pseudo-random generator and a non-zero `upper_bound` to generate a
+    /// random `u16` integer in the range [0, ..., upper_bound - 1]. Note that if the upper
+    /// bound is not a power of two, the distribution will not be completely uniform.
+    public fun next_u16_in_range(random: &mut Random, upper_bound: u16): u16 {
+        assert!(upper_bound > 0, 0);
+        next_u16(random) % upper_bound
+    }
+
+    /// Use the given pseudorandom generator to generate a random `u8`.
+    public fun next_u8(random: &mut Random): u8 {
+        *vector::borrow(&next_digest(random), 0)
+    }
+
+    /// Use the given pseudo-random generator and a non-zero `upper_bound` to generate a
+    /// random `u8` integer in the range [0, ..., upper_bound - 1]. Note that if the upper
+    /// bound is not a power of two, the distribution will not be completely uniform.
+    public fun next_u8_in_range(random: &mut Random, upper_bound: u8): u8 {
+        assert!(upper_bound > 0, 0);
+        next_u8(random) % upper_bound
+    }
+
+    /// Use the given pseudorandom generator to generate a random `bool`.
+    public fun next_bool(random: &mut Random): bool {
+        next_u8(random) % 2 == 1
+    }
+
+    #[test]
+    fun test_next_bytes() {
```

Comment:
We have been putting tests in their own module, but I'll leave that for @emmazzz or @damirka